include/verilated.h

```diff
@@ -314,13 +315,23 @@ class VerilatedModel VL_NOT_FINAL {
 class VerilatedModule VL_NOT_FINAL {
     VL_UNCOPYABLE(VerilatedModule);
 
-private:
-    const char* m_namep;  // Module name
+protected:
+    // Module name, set in emitted constructor, __ prefix to avoid clashing with signal
+    const char* /*const*/ __m_namep;
+
 public:
-    explicit VerilatedModule(const char* namep);  // Create module with given hierarchy name
-    ~VerilatedModule();
-    const char* name() const VL_MT_SAFE_POSTINIT { return m_namep; }  ///< Return name of module
+    VerilatedModule() = default;
+    ~VerilatedModule() = default;
+    const char* name() const VL_MT_SAFE_POSTINIT { return __m_namep; }  ///< Return name of module
 };
+// To avoid many automatically generated constructor and destructor calls
+// in the generated model when there are a large number of modules, the
+// emitted modules should be trivially constructible/destructible. Make sure
+// at least the base class is such.
+static_assert(std::is_trivially_constructible<VerilatedModule>::value,
+              "VerilatedModule should be trivially constructible");
+static_assert(std::is_trivially_destructible<VerilatedModule>::value,
+              "VerilatedModule should be trivially destructible");
```

Comment:
@wsnyder I'm not a big fan of having to do this, but this needs to be default constructible/destructible to avoid the blowup of the Syms constructor/destructor.

Given there is only one member, which is trivial to code generate in `emitC`, can we just delete `VerilatedModule` instead? Anywhere it is used, the concrete subclass is available so today it has no real functional purpose.

Reply:
This was originally to make the code closer to SystemC which has similar base class.  Its ok to remove.